src/main/scala/beam/agentsim/infrastructure/geozone/TopDownEqualDemandsGeoZoneHexGenerator.scala

```diff
@@ -0,0 +1,35 @@
+package beam.agentsim.infrastructure.geozone
+
+class TopDownEqualDemandsGeoZoneHexGenerator private[geozone] (
+  initialContent: IndexedSeq[HexagonLeaf],
+  bucketsGoal: Int,
+  iterationsThreshold: Int = 1000
+) extends GeoZoneHexGenerator {
+
+  private def generateEqualDemandH3Indexes(): Seq[Hexagon[_]] = {
+    var result: IndexedSeq[Hexagon[_]] = initialContent
+    var totalSize = result.map(_.totalNumberOfBuckets).sum
+    var counter = 0
+    while (totalSize < bucketsGoal & counter < iterationsThreshold) {
+      counter += 1
+      val position = TopDownEqualDemandsSplitter.chooseOneToSplit(result, bucketsGoal, numberOfPoints(result))
+      val afterSplit: Seq[HexagonBranch] = result(position).split(bucketsGoal)
+      result = result.patch(position, afterSplit, 1)
```

Comment:
Patch runtime is linear, but I guess it is not a problem here because outer loop is bounded?

Reply:
Nice catch! This is something that can improve performance, but I believe is not significantly in most cases. Few reasons:
- There is a `bucketsGoal` that limits the number of executions
- There is a `iterationsThreshold` that limit the number of executions to `1000`. However the convergence happens in a factor split by 6. Which means starting from only one bucket only `167` iterations will be needed.
- The total execution time for Austin is around 3-4seconds(including generating the shape) on my machine). 

On the other hand if the number of buckets is very huge this is the place to be optimised, but I don´t believe this should be part of this PR